<launch>

    <arg name="use_cable_sensors" default="true"/>
	<arg name="frame" default="data"/>
    <arg name="visualize" default="true"/>
    <arg name="cable_unit_conversion"/>
    <arg name="cable_map_filename"/>
    <arg name="initial_estimate_filename"/>

	<node pkg="estimation" type="state_estimation" name="state_estimation" output="screen">
        <param name="red_endcaps_topic" value="/tensegrity/endcap/red/positions"/>
        <param name="blue_endcaps_topic" value="/tensegrity/endcap/blue/positions"/>
        <param name="green_endcaps_topic" value="/tensegrity/endcap/green/positions"/>
        <param name="tensegrity_pose_topic" value="/tensegrity/poses"/>
        <param name="sensors_topicname" value="/tensegrity/sensors"/>
        <param name="use_cable_sensors" value="$(arg use_cable_sensors)"/>
        <param name="frame" value="$(arg frame)"/>
        <param name="cable_unit_conversion" value="$(arg cable_unit_conversion)"/>
        <param name="cable_map_filename" value="$(arg cable_map_filename)"/>
        <param name="initial_estimate_filename" value="$(arg initial_estimate_filename)"/>
    </node>

    <node if="$(arg visualize)" pkg="nodelet" type="nodelet" name="viz_nodelet"  args="manager" output="screen"/>

    <node if="$(arg visualize)" pkg="nodelet" type="nodelet" name="BarsToMarkersStateEstimation" 
        args="load interface/TensegrityBarsToMarkers viz_nodelet" output="screen">
        <param name="tensegrity_bars_topicname" value="/tensegrity/poses"/>
        <param name="tensegrity_markers_topicname" value="/tensegrity/markers"/>
        <param name="node_id" value="/node/BarsToMarkersStateEstimation"/>
    </node>
</launch>